pkg/doc/verifiable/extension_credential_switch_test.go

```diff
@@ -0,0 +1,232 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package verifiable
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	validCred1 = `
+{
+  "@context": [
+    "https://www.w3.org/2018/credentials/v1",
+    "https://www.w3.org/2018/credentials/examples/ext/type1"
+  ],
+  "id": "http://example.edu/credentials/1872",
+  "type": [
+    "VerifiableCredential",
+    "CredType1"
+  ],
+  "credentialSubject": {
+    "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
+    "s1": "custom subject 1"
+  },
+
+  "c1": "custom field 1",
+
+  "issuer": {
+    "id": "did:example:76e12ec712ebc6f1c221ebfeb1f",
+    "name": "Example University"
+  },
+
+  "issuanceDate": "2010-01-01T19:23:24Z"
+}
+`
+
+	validCred2 = `
+{
+  "@context": [
+    "https://www.w3.org/2018/credentials/v1",
+    "https://www.w3.org/2018/credentials/examples/ext/type2"
+  ],
+  "id": "http://example.edu/credentials/1872",
+  "type": [
+    "VerifiableCredential",
+    "CredType2"
+  ],
+  "credentialSubject": {
+    "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
+    "s2": "custom subject 2"
+  },
+
+  "c2": "custom field 2",
+
+  "issuer": {
+    "id": "did:example:76e12ec712ebc6f1c221ebfeb1f",
+    "name": "Example University"
+  },
+
+  "issuanceDate": "2010-01-01T19:23:24Z"
+}`
+
+	credMissingMandatoryFields = `
+{
+  "@context": [
+    "https://www.w3.org/2018/credentials/v1",
+    "https://www.w3.org/2018/credentials/examples/ext/type2"
+  ],
+  "id": "http://example.edu/credentials/1872",
+  "type": [
+    "VerifiableCredential"
+  ]
+}`
+)
+
+type CustomCredentialProducer struct {
+	Accept func(cred *Credential) bool
```

Comment:
Is there anything about this Accept/Apply pattern that should exist outside of the example/test (as part of the implementation)?

Reply:
Yes, I think so.

Reply:
Ok - please create a GitHub issue with the bits that should be moved into the implementation.

Reply:
Done! #263